physics-based-models-1.ipynb

```diff
@@ -0,0 +1,339 @@
+    "where $g$ is the acceleration of gravity (in absolute value) and $x_0$ is the initial position of the ball (here $'$ denotes the time derivative). For more details on how the model is defined and solved, you can check the PyBaMM examples [Creating a simple ODE model](https://docs.pybamm.org/en/latest/source/examples/notebooks/creating_models/1-an-ode-model.html) and [ODE solver](https://docs.pybamm.org/en/latest/source/examples/notebooks/solvers/ode-solver.html).\n",
+    "\n",
+    "We first define the model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.1.2\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.2\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "%pip install pybamm -q\n",
+    "import pybamm\n",
+    "\n",
+    "# We define the model as a function so we can generate multiple instances of it\n",
+    "def create_model():\n",
+    "    model = pybamm.BaseModel()\n",
+    "\n",
+    "    # Define variables to solve for\n",
+    "    x = pybamm.Variable(\"Position [m]\")\n",
+    "    v = pybamm.Variable(\"Velocity [m.s-1]\")\n",
+    "\n",
+    "    # Define parameters\n",
+    "    g = pybamm.Parameter(\"Gravity acceleration [m.s-2]\")\n",
+    "    x0 = pybamm.Parameter(\"Initial position [m]\")\n",
+    "\n",
+    "    # Define equations and initial conditions\n",
+    "    model.rhs = {x: v, v: -g}\n",
+    "    model.initial_conditions = {x: x0, v: 0}\n",
+    "\n",
+    "    # Define event (collision with ground)\n",
+    "    model.events.append(pybamm.Event(\"Collision with ground\", x))\n",
+    "\n",
+    "    # Define output variables (to plot)\n",
+    "    model.variables = {\n",
+    "        \"Position [m]\": x,\n",
+    "        \"Velocity [m.s-1]\": v,\n",
+    "        \"Acceleration [m.s-2]\": -g,\n",
+    "    }\n",
+    "\n",
+    "    return model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can generate, process, and solve the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = create_model()\n",
+    "\n",
+    "param = pybamm.ParameterValues(\n",
+    "    {\n",
+    "        \"Gravity acceleration [m.s-2]\": 9.81,\n",
+    "        \"Initial position [m]\": 100,\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "param.process_model(model)\n",
+    "disc = pybamm.Discretisation()  # use the default discretisation\n",
```

Comment:
since there are no spatial operators you should be able to skip `disc` and it will happen in the solver (not sure if we do this in the pybamm examples)

Reply:
I tried removing the discretisation, but it fails (due to the event, I suspect).